Documentation improvments

aeslaughter opened the pull request:
I added some features and fixes that I am using to document the level set module, which I will put up a PR for shortly.

* Allows multiple names in \cite commands
* Adds ability to indent, prefix, and suffix text extracted from files
* Adds figure numbers
* Adds equation numbers

(refs #8329)

aeslaughter committed:
Improves bibtex to support list of keys in cite commands

(refs #8329)

aeslaughter committed:
Add prefix, suffix, and indent feature to text extraction

(refs #8329)

aeslaughter committed:
Improve regex for class definition location

(refs #8329)

aeslaughter committed:
Fix return code bug in check.py

(refs #8329)

aeslaughter committed:
Corrected logic for building action lists of groups

(refs #8329)

aeslaughter committed:
Adds ability to have auto numbered figures

(refs #8329)

aeslaughter committed:
Adds per page equation numbers

(refs #8329)

moosebuild commented:
Job `Moose Documentation:linux-clang` on be86c81 wanted to post the following:

View the site at http://mooseframework.org/docs/PRs/8446/site

moosebuild commented:
Job `Moose Documentation:linux-clang` on 5e9fe4e wanted to post the following:

View the site at http://mooseframework.org/docs/PRs/8446/site

aeslaughter committed:
Improve threading behavior to spawn less processes

(refs #8329)

moosebuild commented:
Job `Moose Documentation:linux-clang` on 6966889 wanted to post the following:

View the site at http://mooseframework.org/docs/PRs/8446/site

aeslaughter committed:
Adds actions of objects that exist in a group

This allows applications (i.e., BISON) that do not have framework docs include to get complete systems lists
(refs #8329)

moosebuild commented:
Job `Moose Documentation:linux-clang` on facf69d wanted to post the following:

View the site at http://mooseframework.org/docs/PRs/8446/site

brianmoose commented on docs/content/utilities/documentation/moose_flavored_markdown.md:
```diff
@@ -225,6 +227,56 @@ A full slideshow example might be:
 ---
 
+## Figures
+When writing documentation is customary to reference figures within text by number. To create a numbered figure use
```
it is customary

brianmoose commented on docs/content/utilities/documentation/moose_flavored_markdown.md:
```diff
@@ -225,6 +227,56 @@ A full slideshow example might be:
+!figure docs/media/memory_logger-plot_multi.png width=250px caption=The numbered prefix is automatically applied to the caption. id=fig:memory_logger
+
+First, the caption will automatically be prefixed with the figure number (e.g., Figure \ref{fig:dark_mode}). The
+numbering begins at one and is reset on each page. The prefix "Figure" can be modified with by setting
```
modified by setting

aeslaughter committed:
Address comments

(#8329)

moosebuild commented:
Job `Moose Documentation:linux-clang` on 803b736 wanted to post the following:

View the site at http://mooseframework.org/docs/PRs/8446/site
